### ISSUE TITLE
[@typescript-eslint/camelcase] How to disable?

### ISSUE THREAD

**Issue by bobsilverberg** [typescript-eslint/typescript-eslint#2050]
**Repro**

```JSON
{
  "rules": {
    "camelcase": "off",
    // "@typescript-eslint/camelcase": ["off"]
    "@typescript-eslint/naming-convention": [
      "off",
      {
        "selector": "default",
        "format": null
      }
    ]
  }
}
```

```TS
// your repro code case
my_variable = 0;
```

**Expected Result**
No errors found

**Actual Result**
Error reported `error  Identifier 'my_variable' is not in camel case  @typescript-eslint/camelcase`

**Additional Info**
I don't want any non-camelcase identifiers to trigger an error. My first instinct was to just use `"@typescript-eslint/camelcase": ["off"]`, but the docs say that it is deprecated, and I should use `@typescript-eslint/naming-convention` instead. The config listed above is my attempt to disable the camelcase rule via `@typescript-eslint/naming-convention`. I did see the comment at the bottom of the docs that says "When Not To Use It" : "If you do not want to enforce naming conventions for anything.", but I'm not sure how to disable the rule without using `@typescript-eslint/camelcase"`, which is deprecated.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.18.0` |
| `@typescript-eslint/parser`        | `2.18.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `v12.3.1` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
Likely there's more to your config that's causing the rule to not be disabled.
Have you got a link to the setup?

You can also use [`--print-config`](https://eslint.org/docs/user-guide/command-line-interface#print-config) to help debug what ESLint is resolving your config to.

**Comment by bobsilverberg** [comment]
Thanks @bradzacher. I see now it's coming from `plugin:@typescript-eslint/recommended`, which includes `"@typescript-eslint/camelcase": "error"`, so I guess we need to manually switch that to `off` in order to override that rule locally.
